srcs/Server.cpp

```diff
@@ -10,30 +10,93 @@
 /* ************************************************************************** */
 
 Server::Server() {}
-Server::Server(const std::string& server_block, const std::string& location_blocks)
-: 
+
+Server::Server(ServerManager* server_manager, const std::string& server_block, std::vector<std::string>& location_blocks, Config* config)
 {
-	this->m_server_name.clear();
-	this->m_host.clear();
-	this->m_port = 0;
-	this->m_fd = 0;
-	this->m_request_uri_limit_size = 0;
-	this->m_request_header_limit_size = 0;
-	this->m_limit_client_body_size = 0;
-	this->m_default_error_page.clear();
+	std::map<std::string, std::string> server_map = ft::stringVectorToMap(ft::split(server_block, '\n'), ' ');
+	struct sockaddr_in server_addr;
+	std::string uri;
+	std::string block;
+	std::string error_page;
+	std::string line;
+
+	m_manager = server_manager;
+	m_config = config;
+	if (ft::hasKey(server_map, "server_name"))
+		m_server_name = server_map["server_name"];
+	else
+		m_server_name = "noname";
+	m_host = server_map["host"];
+	m_port = std::stoi(server_map["port"]);
+	m_request_uri_limit_size = std::stoi(server_map["REQUEST_URI_LIMIT_SIZE"]);
+	m_request_header_limit_size = std::stoi(server_map["REQUEST_HEADER_LIMIT_SIZE"]);
+	m_limit_client_body_size = std::stoi(server_map["LIMIT_CLIENT_BODY_SIZE"]);
+	error_page = server_map["DEFAULT_ERROR_PAGE"];
+	//socket 생성
+	if((m_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) == -1)
+		throw "SOCKET ERROR";
+	int value = true;
+	if (setsockopt(m_fd, SOL_SOCKET, SO_REUSEADDR, &value, sizeof(int)) == -1)
+		throw "SOCKET_OPTION ERROR";
+	bzero(&server_addr, sizeof(struct sockaddr_in));
+	// memset((void *)&server_addr, 0x00, sizeof(server_addr));
+	server_addr.sin_family = AF_INET;
+	server_addr.sin_addr.s_addr = inet_addr(m_host.c_str());
+	server_addr.sin_port = htons(8080);
```

Comment:
```suggestion
	server_addr.sin_port = htons(m_port);
```

8080번 이외의 포트에서도 연결을 가능케해야 하지 않을까요?